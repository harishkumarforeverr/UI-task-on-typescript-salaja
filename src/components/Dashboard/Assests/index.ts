import DashboardImageCircut from "./DashboardImageCircut.png";
import nextIcon_pro from "./nextIcon_pro.png";
import Ellipse from "./Ellipse.png";
import cardImage_1 from "./cardImage_1.png"; 
import cardImage_2 from "./cardImage_2.png";
import cardImage_3 from "./cardImage_3.png";
import cardImage_4 from "./cardImage_4.png";
import cardImage_6 from "./cardImage_6.png";
import cardImage_5 from "./cardImage_5.png";
 

export const Assests = {
  DashboardImageCircut,nextIcon_pro,Ellipse,cardImage_1
  
  ,cardImage_2,cardImage_3,cardImage_4,cardImage_5,cardImage_6};
