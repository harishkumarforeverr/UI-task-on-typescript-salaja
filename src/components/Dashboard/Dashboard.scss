.dashboard {
  display: flex;
  justify-content: space-between;
  margin-right: 20px;
  margin-top: 20px;
  gap: 37px;
  padding-left: 2%;
  height: 52rem;
  margin-bottom: 3rem;
  overflow: hidden;
  overflow-y: hidden;
  padding-bottom: 2rem;
  .lightRed {
    color: white;
  }
  .ConatinersIconsTitle {
    padding: 1%;
    display: grid;
    gap: 40px;
    width: 100%;
    height: 100%;
  }
  .cardsRigghtContainer {
    display: grid;
    // display: flex;
    // flex-wrap: wrap;
    margin-left: 10px;
    grid-template-columns: 1fr 1fr 1fr !important;
    height: 100%;
    gap: 20px;
    .conatinerCardImage {
      cursor: pointer;
      display: grid;
      justify-content: center;
      align-items: center;
      box-sizing: border-box;

      width: unset !important;
      min-height: 8.5 !important;

      background:  #F2F8FB !important;
      box-shadow: 0px 6px 12px -2px rgba(0, 0, 0, 0.2) !important;
      border-radius: 6px !important;
    }
    .rightConatinesIcons {
      display: grid;
      justify-content: center;
      align-items: center;
    }
    .rightConatinesIcons .anticon {
      background: #fac7c7;

      display: flex;
      justify-content: center;
      align-items: center;
      border-radius: 50%;
      color: rgba(0, 0, 0, 0.489);

      font-size: 1.3rem;
      margin-bottom: 1rem;
      width: 45px !important;
      height: 45px !important;
    }
    .titleRightConatiner {
      font-family: 'Helvetica';
      font-style: normal;
      font-weight: 400;
      font-size: 20px;
      line-height: 21px;
      margin-top: 4px;
      color: #1C99C4;
      white-space: nowrap;
      &:hover{
        text-decoration: underline;
      }
    }
  }

  .knowYourDevice {
    box-sizing: border-box;
    // min-width: 529px;
    // height: 662px;
    height: 100%;
    padding-bottom: 1.1%;
    // flex-basis: 45%;
    flex-basis: 60%;
    margin-top: auto;
    margin-top: 1rem;

    min-height: 41.3rem;
    height: 100%;
    background: #f2f8fb;
    box-shadow: 0px 6px 12px -2px rgba(0, 0, 0, 0.2);
    border-radius: 6px;
    left:817px;
    .knowYourDevice_head {
      box-sizing: border-box;
      // min-width: 529px;
      min-height: 48px;
      background: #d4e5ee;
      border-bottom: 2px solid #1c99c4;
      border-radius: 6px 6px 0px 0px;
      display: flex;
      align-items: center;
      padding-left: 20px;
      h1 {
        // min-width: 121px;
        min-height: 18px;
        white-space: nowrap;
        font-family: Helvetica;
        font-style: normal;
        font-weight: 700;
        font-size: 14px;
        letter-spacing: 0.04em;

        color: #1c99c4;
      }
    }
    .knowYourDevice_context {
      // margin-top: 20px;
      display: grid;
      justify-content: center;
      align-items: center;
      .imageKnowContainer{
        display: grid;
        justify-content: center;
        align-items: center;
        // min-width: 469px;
        // min-height: 301px;
        // padding: 2rem;
        margin: 1rem 0;
      }
      img {
        width: 93%;
        margin: 0 auto;
      
      }
      p {
        width: 90%;
        margin: 0 auto;
        // margin-top: 20px;
        // min-width: 469px;
        min-height: 167px;

        font-family: Helvetica;
        font-style: normal;
        font-weight: 400;
        font-size: 14px;
        line-height: 2rem;
        text-align: justify;
        letter-spacing: 0.04em;
        font-family: Roboto;

        color: #231f20;
      }
    }
  }
}

.GetImages{
  position: relative;
  .ellipse_image{
    width: 4rem;
  }
  .GetImages_child{
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }
  .image_inner{
    width: 2.5rem;
height: 2.5rem;
  }
}