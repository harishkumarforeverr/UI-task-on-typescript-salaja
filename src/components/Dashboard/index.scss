.appConatiner_DashboardConatiner {
  .appConatiner_drawer {
    position: fixed;
    right: 0;
    top: 2rem;
  }
  .i2cControl,
  .i2cControlFalse {
    cursor: pointer;
    width: 36px;
    height: 54px;

    background: #ffffff;
    border-radius: 8px 0px 0px 8px;
    display: flex;
    flex-direction: column;
    gap: 0.5rem;
    padding-top: 0.4rem;
    .rightIcon {
      font-size: 0.81rem;
    }
    .i2cLabel {
      width: 19px;
      height: 12px;
      font-family: "Helvetica";
      font-style: normal;
      font-weight: 700;
      font-size: 12px;
      line-height: 12px;
      color: #1c99c4;

      p {
        writing-mode: vertical-lr;
      }
    }
  }
  .i2cControlFalse {
    width: 36px;
    height: 86px;
  }
}
