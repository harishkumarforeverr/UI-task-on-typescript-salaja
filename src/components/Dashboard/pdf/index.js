const link1 = require("./pdf_file_1.pdf");
const link4 = require("./pdf_file_2.pdf");
const link3 = require("./pdf_file_3.pdf");

const link2 = "https://www.ti.com/product/MCF8316A";
const link5 = "https://www.ti.com/tool/MCF8316AEVM";
const link6 =
  "https://e2e.ti.com/support/motor-drivers-group/motor-drivers/f/motor-drivers-forum";

export const usersLinks = {
  link1,
  link2,
  link3,
  link4,
  link5,
  link6,
};
