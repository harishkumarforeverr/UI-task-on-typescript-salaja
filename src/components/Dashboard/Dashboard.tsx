import React, { useState } from "react";
import {
  UserOutlined,
  SettingOutlined,
  ProjectOutlined,
  PullRequestOutlined,
  ExperimentOutlined,
  TransactionOutlined,
} from "@ant-design/icons";
import "./Dashboard.scss";
import { paths } from "../../utils/constant";
import { useNavigate } from "react-router-dom";
import { Assests } from "./Assests";
import Select, { SelectChangeEvent } from "@mui/material/Select";
import { Card } from "antd";
import SelectWrapper from "../common/SelectWrapper";
function Dashboard({ setView }: { setView: any }) {
  // const [age, setAge] = React.useState<string>("");

  // const handleChange = (event: SelectChangeEvent) => {
  //   // setAge(event.target.value);
  // };
  const [motor, setMotor] = useState("");
  const navigate = useNavigate();
  const handleTheMotor = () => {
    console.log(motor);
    if (motor !== "") {
      setView("QuickOptimizationWidget");
    }
  };
  const userInfoObj = [
    {
      title: "Users's Guide",
      icons: <UserOutlined />,
      link: paths.UserGuidesPage,
    },
    {
      title: "Product Page",
      icons: <ExperimentOutlined />,
      link: "/",
    },
    {
      title: "Data Sheet",
      icons: <ProjectOutlined />,
      link: "/",
    },
    {
      title: "Tuning Page",
      icons: <SettingOutlined />,
      link: "/",
    },
    {
      title: "Tools Page",
      icons: <PullRequestOutlined />,
      link: "/",
    },
    {
      title: "E2E testing",
      icons: <TransactionOutlined />,
      link: "/",
    },
  ];
  const handleChange = (event: SelectChangeEvent) => {
    setMotor(event.target.value);
  };
  return (
    <div className="dashboard">
      <div className="ConatinersIconsTitle">
        <div>
          <SelectWrapper
            handleChange={handleChange}
            handleTheMotor={handleTheMotor}
            BtnSelectDeviceOpacity="1"
            dotColor="#cc0000"
            deviceTitle="MCF8316EVM"
            devDesc="This GUI supports MCF8316EVM"
            // devSubDesc="Correct device detected : MCF8316A"
          />
        </div>
        <div className="cardsRigghtContainer">
          {userInfoObj.map((obj) => {
            return (
              <Card
                className="conatinerCardImage"
                bordered={false}
                style={{ width: 300 }}
                onClick={() => {
                  navigate(obj.link);
                }}
              >
                <div className="rightConatinesIcons">{obj.icons}</div>
                <div className="titleRightConatiner">{obj.title}</div>
              </Card>
            );
          })}
        </div>
      </div>
      {/* <div> */}
      <div className="knowYourDevice">
        <div className="knowYourDevice_head">
          <h1>Know Your Device</h1>
        </div>
        <div className="knowYourDevice_context">
          <img src={Assests.DashboardImageCircut} alt="Assests" />
          <div>
            <p>
              The MCF8316A is a 4.5-V to 35-V, 8-A peak integrated three-phase
              gate driver IC with sensorless field-oriented control for motor
              drive applications. It provides three accurately trimmed and
              temperature compensated halfbridge MOSFETS, gate drivers, charge
              pump, current sense amplifier, linear regulator for the external
              load and adjustable buck regulator. The I2C interface variant
              (MCF8316A) also provides a standard I2C interface for configuring
              the various device settings and reading fault diagnostic
              information through an external controller.
            </p>
          </div>
        </div>
      </div>
      {/* </div> */}
    </div>
  );
}

export default Dashboard;
