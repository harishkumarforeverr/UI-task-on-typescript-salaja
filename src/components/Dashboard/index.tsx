import React, { useState } from "react";
import Dashboard from "./Dashboard";

import "./index.scss";
import QuickOptimizationWidget from "./QuickOptimizationWidget/QuickOptimizationWidget";
function DashboardContainer() {
  const [view, setView] = useState("Dashboard");
  const [open, setOpen] = useState(false);
  return (
    <>
      {view == "Dashboard" && <Dashboard setView={setView} />}
      {view == "QuickOptimizationWidget" && (
        <QuickOptimizationWidget setView={setView} />
      )}
    </>
  );
}

export default DashboardContainer;
