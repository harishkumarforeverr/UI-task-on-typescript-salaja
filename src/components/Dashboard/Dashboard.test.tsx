import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer"; 
import Dashboard from "./Dashboard";

const mockHistoryPush = jest.fn();
const mockGoBack = jest.fn();

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "/",
  }),
  useHistory: () => ({
    push: mockHistoryPush,
    goBack: mockGoBack,
  }),
  useParams: () => ({
    pathParam: "haribusine-HOAoFA",
  }),
  useNavigate: () => ({
    pathParam: "haribusine-HOAoFA",
  }), 
}));

const setView=jest.fn((arg:any)=>arg)
test('renders the compoenet and finding whether a text present in the dom or not', () => {
  const MyApp=()=>{
    return (
      <Router>
        <Dashboard setView={setView} />
      </Router>
    )
  }
  render(<MyApp />);
  const linkElement = screen.getByText("Users's Guide");
  expect(linkElement).toBeInTheDocument();
}); 

it("creating the snapshot of the compoenent correctly", () => {
  const MyApp = () => {
    return (
      <Router>
        <Dashboard setView={setView} />
      </Router>
    );
  };
  const tree = renderer.create(<MyApp />).toJSON();
  expect(tree).toMatchSnapshot();
});
