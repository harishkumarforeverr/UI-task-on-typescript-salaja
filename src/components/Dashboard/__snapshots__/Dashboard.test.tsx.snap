// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creating the snapshot of the compoenent correctly 1`] = `
<div
  className="dashboard"
>
  <div
    className="ConatinersIconsTitle"
  >
    <div>
      <div
        className="ant-card conatinerCard css-dev-only-do-not-override-1n7nwfa"
        style={
          Object {
            "width": 300,
          }
        }
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div>
            <div
              className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink={true}
                id="selectlabelCommon"
              >
                Select Device*
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-4gwozz-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                onClick={[Function]}
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="selectlabelCommon demo-simple-select"
                  className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="demo-simple-select"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  Select Your Device
                </div>
                <input
                  aria-hidden={true}
                  className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onChange={[Function]}
                  required={false}
                  tabIndex={-1}
                  value="Select Your Device"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-14lo706"
                  >
                    <span>
                      Select Device*
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div>
            <div
              className="deatiles"
            >
              <span
                className="ColorContainer"
                style={
                  Object {
                    "background": "#cc0000",
                  }
                }
              />
              <span
                className="numberText"
              >
                MCF8316EVM
              </span>
            </div>
            <p
              className="GuiSupport"
            >
              This GUI supports MCF8316EVM
            </p>
            <p
              className="GuiSupport"
              style={
                Object {
                  "fontSize": "12px",
                  "opacity": "0.8",
                }
              }
            />
          </div>
          <button
            className="ant-btn css-dev-only-do-not-override-1n7nwfa ant-btn-default lightRed btn"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "opacity": "1",
              }
            }
            type="button"
          >
            <span>
              Select Device
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="cardsRigghtContainer"
    >
      <div
        className="ant-card conatinerCardImage css-dev-only-do-not-override-1n7nwfa"
        onClick={[Function]}
        style={
          Object {
            "width": 300,
          }
        }
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="rightConatinesIcons"
          >
            <div
              className="GetImages"
            >
              <img
                className="ellipse_image"
                src="Ellipse.png"
              />
              <div
                className="GetImages_child"
              >
                <img
                  alt="image_inner"
                  className="image_inner"
                  src="cardImage_1.png"
                />
                 
              </div>
            </div>
          </div>
          <div
            className="titleRightConatiner"
          >
            Users's Guide
          </div>
        </div>
      </div>
      <div
        className="ant-card conatinerCardImage css-dev-only-do-not-override-1n7nwfa"
        onClick={[Function]}
        style={
          Object {
            "width": 300,
          }
        }
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="rightConatinesIcons"
          >
            <div
              className="GetImages"
            >
              <img
                className="ellipse_image"
                src="Ellipse.png"
              />
              <div
                className="GetImages_child"
              >
                <img
                  alt="image_inner"
                  className="image_inner"
                  src="cardImage_2.png"
                />
                 
              </div>
            </div>
          </div>
          <div
            className="titleRightConatiner"
          >
            Product Page
          </div>
        </div>
      </div>
      <div
        className="ant-card conatinerCardImage css-dev-only-do-not-override-1n7nwfa"
        onClick={[Function]}
        style={
          Object {
            "width": 300,
          }
        }
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="rightConatinesIcons"
          >
            <div
              className="GetImages"
            >
              <img
                className="ellipse_image"
                src="Ellipse.png"
              />
              <div
                className="GetImages_child"
              >
                <img
                  alt="image_inner"
                  className="image_inner"
                  src="cardImage_3.png"
                />
                 
              </div>
            </div>
          </div>
          <div
            className="titleRightConatiner"
          >
            Data Sheet
          </div>
        </div>
      </div>
      <div
        className="ant-card conatinerCardImage css-dev-only-do-not-override-1n7nwfa"
        onClick={[Function]}
        style={
          Object {
            "width": 300,
          }
        }
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="rightConatinesIcons"
          >
            <div
              className="GetImages"
            >
              <img
                className="ellipse_image"
                src="Ellipse.png"
              />
              <div
                className="GetImages_child"
              >
                <img
                  alt="image_inner"
                  className="image_inner"
                  src="cardImage_4.png"
                />
                 
              </div>
            </div>
          </div>
          <div
            className="titleRightConatiner"
          >
            Tuning Page
          </div>
        </div>
      </div>
      <div
        className="ant-card conatinerCardImage css-dev-only-do-not-override-1n7nwfa"
        onClick={[Function]}
        style={
          Object {
            "width": 300,
          }
        }
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="rightConatinesIcons"
          >
            <div
              className="GetImages"
            >
              <img
                className="ellipse_image"
                src="Ellipse.png"
              />
              <div
                className="GetImages_child"
              >
                <img
                  alt="image_inner"
                  className="image_inner"
                  src="cardImage_5.png"
                />
                 
              </div>
            </div>
          </div>
          <div
            className="titleRightConatiner"
          >
            Tools Page
          </div>
        </div>
      </div>
      <div
        className="ant-card conatinerCardImage css-dev-only-do-not-override-1n7nwfa"
        onClick={[Function]}
        style={
          Object {
            "width": 300,
          }
        }
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="rightConatinesIcons"
          >
            <div
              className="GetImages"
            >
              <img
                className="ellipse_image"
                src="Ellipse.png"
              />
              <div
                className="GetImages_child"
              >
                <img
                  alt="image_inner"
                  className="image_inner"
                  src="cardImage_6.png"
                />
                 
              </div>
            </div>
          </div>
          <div
            className="titleRightConatiner"
          >
            E2E testing
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="knowYourDevice"
  >
    <div
      className="knowYourDevice_head"
    >
      <h1>
        Know Your Device
      </h1>
    </div>
    <div
      className="knowYourDevice_context"
    >
      <div
        className="imageKnowContainer"
      >
        <img
          alt="Assests"
          src="DashboardImageCircut.png"
        />
      </div>
        
      <div>
        <p>
          The MCF8316A is a 4.5-V to 35-V, 8-A peak integrated three-phase gate driver IC with sensorless field-oriented control for motor drive applications. It provides three accurately trimmed and temperature compensated halfbridge MOSFETS, gate drivers, charge pump, current sense amplifier, linear regulator for the external load and adjustable buck regulator. The I2C interface variant (MCF8316A) also provides a standard I2C interface for configuring the various device settings and reading fault diagnostic information through an external controller.
        </p>
      </div>
    </div>
  </div>
</div>
`;
